packages/controllers/src/controllers/ChainController.ts

```diff
@@ -767,6 +818,48 @@ const controller = {
 
   getLastConnectedSIWECaipNetwork(): CaipNetwork | undefined {
     return state.lastConnectedSIWECaipNetwork
+  },
+  async fetchTokenBalance(onError?: (error: unknown) => void): Promise<Balance[]> {
```

Comment:
Is this just moving things or new function?

Reply:
moving from AccController. TBCH this shold not go here, probably best as utility